api/v1alpha1/secretagentconfiguration_webhook.go

```diff
@@ -20,8 +20,72 @@ import (
 	"fmt"
 
 	"github.com/go-playground/validator/v10"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook"
 )
 
+// log is for logging in this package.
+var log = logf.Log.WithName("secretagentconfiguration-webhook")
+
+//SetupWebhookWithManager registers the webhook with the manager
+func (r *SecretAgentConfiguration) SetupWebhookWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewWebhookManagedBy(mgr).
+		For(r).
+		Complete()
+}
+
+// +kubebuilder:webhook:path=/mutate-secret-agent-secrets-forgerock-io-v1alpha1-secretagentconfiguration,mutating=true,failurePolicy=fail,groups=secret-agent.secrets.forgerock.io,resources=secretagentconfigurations,verbs=create;update,versions=v1alpha1,name=msecretagentconfiguration.kb.io
+
+var _ webhook.Defaulter = &SecretAgentConfiguration{}
+
+// Default implements webhook.Defaulter so a webhook will be registered for the type
+func (r *SecretAgentConfiguration) Default() {
+	// log.Info("default", "name", r.Name)
+	// TODO LATER: fill in defaulting logic here if we ever want to.
+	//example
+	// if r.Spec.AppConfig.SecretsManager == "" {
+	// 	r.Spec.AppConfig.SecretsManager = SecretsManagerNone
+	// }
+}
+
+// change verbs to "verbs=create;update;delete" if you want to enable deletion validation.
+// +kubebuilder:webhook:verbs=create;update,path=/validate-secret-agent-secrets-forgerock-io-v1alpha1-secretagentconfiguration,mutating=false,failurePolicy=fail,groups=secret-agent.secrets.forgerock.io,resources=secretagentconfigurations,versions=v1alpha1,name=vsecretagentconfiguration.kb.io
+
+var _ webhook.Validator = &SecretAgentConfiguration{}
+
+// ValidateCreate implements webhook.Validator so a webhook will be registered for the type
+func (r *SecretAgentConfiguration) ValidateCreate() error {
+	log.Info("Validating new SecretAgentConfiguration", "name", r.Name)
+	return r.ValidateSecretConfiguration()
+}
+
+// ValidateUpdate implements webhook.Validator so a webhook will be registered for the type
+func (r *SecretAgentConfiguration) ValidateUpdate(old runtime.Object) error {
+	log.Info("Validating existing SecretAgentConfiguration", "name", r.Name)
+	return r.ValidateSecretConfiguration()
+}
+
+// ValidateDelete implements webhook.Validator so a webhook will be registered for the type
+func (r *SecretAgentConfiguration) ValidateDelete() error {
+	//We're not using this function atm. Keeping it here in case we do want to start using it later.
+	log.Info("validate delete", "name", r.Name)
+	return nil
+}
+
+//ValidateSecretConfiguration Validates the SecretAgentConfiguration object
+func (r *SecretAgentConfiguration) ValidateSecretConfiguration() error {
```

Comment:
We can. I updated other functions in the controller before. In hindsight, I think we should have left `r` in the controller as that's the standard in the kubebuilder. I am inclined to even change the controller's `reconciler` back to `r`:

```
func (>>THIS>>reconciler<<THIS<< *SecretAgentConfigurationReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
```

Reply:
Yeah... As I think of this, I think we should keep the standard naming of things. After all, this is a public project. It's beneficial to stick to standards. I won't change it now, but I suggest to go back to `r` in the controller's file as well.

Reply:
Here I think `r` represents a `SecretAgentConfiguration`, not a reconciler? How does using `r` help us know it's a `SecretAgentConfiguration`?

Reply:
I'm not convinced we're meant to follow kubebuilder's docs to a tee, like in variable naming. I think they're in a hurry and trying to provide loose examples. I often find package documentation more useful than website docs. Anyway.

Reply:
In the controller it represents a reconciler, so `r` makes more sense. Here, it represents our custom object. There's no docs to compare to on that.

Looking through the controller, and how many places `reconciler` is used throughout long functions, I think it would be incredibly annoying to have to be looking at some code and wonder what `r` is and have to scroll to the top of the function to find out. `reconciler` reads way better and makes intuitive sense.